lib/doorkeeper-sequel/mixins/access_token_mixin.rb

```diff
@@ -104,6 +110,7 @@ def scopes_match?(token_scopes, param_scopes, app_scopes)
       end
 
       def authorized_tokens_for(application_id, resource_owner_id)
+        resource_owner_id = resource_owner_id&.id if polymorphic_resource_owner? && !resource_owner_id.is_a?(Integer)
         where(application_id: application_id,
               resource_owner_id: resource_owner_id,
```

Comment:
@nbulaj, should we check type if its polymorphic? Tests are passed but what if an user provides an id and type but this id have 2 times with different types?

Example:
```
===========
resource_owner_id   resource_owner_type
     1                 User
     1                 Admin
===========
```

Reply:
Yes, we have. I checked on active_record orm library for it. I will add it now.

Reply:
Done.